lib/influxdb/client/http.rb

```diff
@@ -84,8 +85,18 @@ def resolve_error(response)
       raise InfluxDB::Error, response
     end
 
-    def handle_successful_response(response, options)
-      parsed_response = JSON.parse(response.body) if response.body
+    def handle_successful_response(response, options, json_streaming)
+      if json_streaming
+        if response.body
+          parsed_response = {}
+          response.body.each_line do |line|
+            parsed_response.merge!(JSON.parse(line)) { |_key, old, new| old + new }
+          end
+        end
+      elsif response.body
+        parsed_response = JSON.parse(response.body)
+      end
```

Comment:
Nice. I've only been Ruby coding for 6 months, so .with_object() was the key I was missing. I've adapted your suggestion.